linker/shared/src/main/scala/org/scalajs/linker/analyzer/Infos.scala

```diff
@@ -578,9 +578,27 @@ object Infos {
             case CreateJSClass(className, _) =>
               builder.addInstantiatedClass(className)
 
-            case Transient(CallHelper(_, args)) =>
+            case Transient(t) =>
               // This should only happen when called from the Refiner
-              args.foreach(traverse)
+
+              t match {
+                case SystemArrayCopy(src, srcPos, dest, destPos, length) =>
+                  traverse(src); traverse(srcPos); traverse(dest); traverse(destPos); traverse(length)
+                case ZeroOf(runtimeClass) =>
+                  traverse(runtimeClass)
+                case NativeArrayWrapper(elemClass, nativeArray) =>
+                  traverse(elemClass); traverse(nativeArray)
+                case NumberOfLeadingZeroes(num) =>
+                  traverse(num)
+                case ObjectClassName(obj) =>
+                  traverse(obj)
+                case Array2TypedArray(expr, primRef) =>
+                  traverse(expr)
+                case TypedArray2Array(expr, primRef) =>
+                  traverse(expr)
+                case _ =>
+                  throw new IllegalArgumentException("unexpected transient: " + t)
+              }
```

Comment:
@sjrd this makes me think: Maybe we should expect transients to also implement transformation and traversal?

Reply:
Yes, that is a good point. I was also thinking that the `TransientValue` should carry its `tpe: Type`, and that the `Transient` would derive its `tpe` from its value. I found it annoying to have to pass a computed `tpe` to the constructor of `Transient`, when in fact that was entirely determined by the underlying `TransientValue`.